geoalchemy2/admin/dialects/mysql.py

```diff
@@ -31,16 +31,28 @@ def reflect_geometry_column(inspector, table, column_info):
     column_name = column_info.get("name")
     schema = table.schema or inspector.default_schema_name
 
-    # Check geometry type, SRID and if the column is nullable
-    geometry_type_query = """SELECT DATA_TYPE, SRS_ID, IS_NULLABLE
-        FROM INFORMATION_SCHEMA.COLUMNS
-        WHERE TABLE_NAME = '{}' and COLUMN_NAME = '{}'""".format(
-        table.name, column_name
-    )
-    if schema is not None:
-        geometry_type_query += """ and table_schema = '{}'""".format(schema)
-    geometry_type, srid, nullable_str = inspector.bind.execute(text(geometry_type_query)).one()
-    is_nullable = str(nullable_str).lower() == "yes"
+    if inspector.dialect.name == "mariadb":
+        geometry_type_query = """SELECT DATA_TYPE, IS_NULLABLE
+            FROM INFORMATION_SCHEMA.COLUMNS
+            WHERE TABLE_NAME = '{}' and COLUMN_NAME = '{}'""".format(
+            table.name, column_name
+        )
+        if schema is not None:
+            geometry_type_query += """ and table_schema = '{}'""".format(schema)
+        geometry_type, nullable_str = inspector.bind.execute(text(geometry_type_query)).one()
+        is_nullable = str(nullable_str).lower() == "yes"
+        srid = 0
+    else:
+        # Check geometry type, SRID and if the column is nullable
+        geometry_type_query = """SELECT DATA_TYPE, SRS_ID, IS_NULLABLE
+            FROM INFORMATION_SCHEMA.COLUMNS
+            WHERE TABLE_NAME = '{}' and COLUMN_NAME = '{}'""".format(
+            table.name, column_name
+        )
+        if schema is not None:
+            geometry_type_query += """ and table_schema = '{}'""".format(schema)
+        geometry_type, srid, nullable_str = inspector.bind.execute(text(geometry_type_query)).one()
+        is_nullable = str(nullable_str).lower() == "yes"
```

Comment:
```suggestion
    if inspector.dialect.name == "mariadb":
        select_srid = "0, "
    else:
        select_srid = "SRS_ID, "

    # Check geometry type, SRID and if the column is nullable
    geometry_type_query = """SELECT DATA_TYPE, {}IS_NULLABLE
        FROM INFORMATION_SCHEMA.COLUMNS
        WHERE TABLE_NAME = '{}' and COLUMN_NAME = '{}'""".format(
        select_srid, table.name, column_name
    )
    if schema is not None:
        geometry_type_query += """ and table_schema = '{}'""".format(schema)
    geometry_type, srid, nullable_str = inspector.bind.execute(text(geometry_type_query)).one()
    is_nullable = str(nullable_str).lower() == "yes"
```
I think it's simpler like this, WDYT?

Reply:
I don't mind either way and it's your code.

Reply:
Ahah ok, thanks :)
Do you have some time to work on a test for this soon? Just to know if I have to find some time on my side.